tools/test/test_depsolve.py

```diff
@@ -261,19 +276,22 @@ def config_combos(tmp_path, servers):
 
 
 @pytest.mark.parametrize("test_case", test_cases)
-@pytest.mark.parametrize("dnf_cmd, detect_fn", [
-    ("./tools/osbuild-depsolve-dnf", has_dnf),
-    ("./tools/osbuild-depsolve-dnf5", has_dnf5),
+@pytest.mark.parametrize("dnf_config, detect_fn", [
+    (None, has_dnf),
+    ('{"use_dnf5": false}', has_dnf),
+    ('{"use_dnf5": true}', has_dnf5),
 ])
-def test_depsolve(tmp_path, repo_servers, dnf_cmd, detect_fn, test_case):
-    if not detect_fn():
-        pytest.skip(f"cannot import support for {dnf_cmd}")
+def test_depsolve(tmp_path, repo_servers, dnf_config, detect_fn, test_case):
+    try:
+        detect_fn()
+    except RuntimeError as e:
+        pytest.skip(e)
 
     pks = test_case["packages"]
 
     for repo_configs, root_dir, opt_metadata in config_combos(tmp_path, repo_servers):
         with TemporaryDirectory() as cache_dir:
-            res = depsolve(pks, repo_configs, root_dir, cache_dir, opt_metadata, dnf_cmd)
+            res = depsolve(pks, repo_configs, root_dir, cache_dir, dnf_config, opt_metadata)
```

Comment:
While looking at this I was wondering if we can/should ensure that the right `depsolver` is picked. It seems useful to ensure that when we ask `depsolve()` to test dnf5 that we somehow know that it DTRT, i.e. a (hypothetical) bug in the way the config is setup/read might lead to the test using the defaults instead of dnf5 and AFAICT we would not notice (sorry, that is not very coherent, I hope it still makes sense). So I wonder if we could/should do something like:
```python
diff --git a/osbuild/solver/dnf.py b/osbuild/solver/dnf.py
index 0d320303..d15d5d03 100755
--- a/osbuild/solver/dnf.py
+++ b/osbuild/solver/dnf.py
@@ -304,6 +304,7 @@ class DNF(SolverBase):
                 "sslclientcert": repo.sslclientcert,
             }
         response = {
+            "solver": "dnf",
             "packages": packages,
             "repos": repositories,
         }
diff --git a/osbuild/solver/dnf5.py b/osbuild/solver/dnf5.py
index 75c20cf1..01b0f7a0 100755
--- a/osbuild/solver/dnf5.py
+++ b/osbuild/solver/dnf5.py
@@ -429,6 +429,7 @@ class DNF5(SolverBase):
                 "sslcacert": get_string_option(repo_cfg.get_sslcacert_option()),
             }
         response = {
+            "solver": "dnf5",
             "packages": packages,
             "repos": repositories,
         }
diff --git a/tools/test/test_depsolve.py b/tools/test/test_depsolve.py
index f0fa713c..7a397353 100644
--- a/tools/test/test_depsolve.py
+++ b/tools/test/test_depsolve.py
@@ -305,3 +305,10 @@ def test_depsolve(tmp_path, repo_servers, dnf_config, detect_fn, test_case):
                 assert n_filelist_files == len(REPO_PATHS)
             else:
                 assert n_filelist_files == 0
+
+            if detect_fn == has_dnf5:
+                assert res["solver"] == "dnf5"
+            elif detect_fn == has_dnf:
+                assert res["solver"] == "dnf"
+            else:
+                raise Exception("unknown detect_fn")
```
including the solver in the response might be useful in general for logging purposes too. Note that the abuse of the detect_fn in the example above is not great but I wanted to provide a quick strawman to illustrate the idea.

Reply:
So, I actually looked into doing that but the response gets processed into a Go struct and it doesn't like it when you add an unknown field. I want to keep this compatible with the existing depsolver for now, but adding something like that is a good idea to work on later.

Reply:
Thanks, that is an excellent point! I opened https://github.com/osbuild/images/compare/main...mvo5:depsolve-result-add-solver?expand=1 with a first draft to make images support this.